JsonLd/Event/ContextBuilderEvent.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Dunglas\ApiBundle\JsonLd\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+class ContextBuilderEvent extends Event
+{
+    private $context;
+
+    public function __construct(array $context) {
+
+        $this->context = $context;
+    }
+
+    /**
+     * @return array
+     */
+    public function getContext()
+    {
+        return $this->context;
+    }
+
+    /**
+     * @param array $context
+     *
+     * @return array
+     */
+    public function setContext(array $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * @param $key
+     * @param $value
+     *
+     * @return ContextBuilderEvent
+     */
+    public function addToContext($key, $value)
```

Comment:
I've a problem with the `setContext` and `addToContext` methods in an event. It's maybe not the "responsibility" of this PR to introduce theses changes that but I think that it'd be far easier to manage with `Context` object instead of the array. That way, you can have an `add` method on it without any problem and the manipulation of context is far more easy (and makes more sense imo):
```php
$event->getContext()->add('foo', 'bar');
```

Instead of:
```
$context = $event->getContext();
$context['foo'] = 'bar';
$event->setContext($context);
```